### PR TITLE
Transaction tool reconstruct

### PR DESCRIPTION
**Pull request by bird512**

### CODE REVIEW

File: core/simple_container.go [hunk 1]
```diff
@@ -17,44 +17,223 @@ package core
...
-
-	return c.txs.Len()
-}
-
-func (c *simpleContainer) sort() {
-	c.lock.Lock()
-	defer c.lock.Unlock()
-
-	sort.Sort(c.txs)
+	return c.pending.size + len(c.queue)
```
Comment (darren0718) [inline core/simple_container.go:234]: should simply return len(txsMap) ?

File: core/simple_container.go [hunk 2]
```diff
@@ -71,41 +250,64 @@ func (c *simpleContainer) get(key common.Hash) *types.Transaction {
...
 	}
+	stateNonce := c.getStateNonce(tx)
+	if !IsTestTransaction(tx) && (tx.Nonce <= stateNonce || tx.Nonce > stateNonce+1000) {
+		_ = fmt.Errorf("nonce error:%v %v", tx.Nonce, stateNonce)
+		return
+	}
+	//check balance before push to pool
+	gas, err := intrinsicGas(tx)
+	if err != nil {
+		_ = fmt.Errorf("discard transaction with gas not enough for gas limit")
```
Comment (darren0718) [inline core/simple_container.go:292]: error must be handled or returned!

**Review by darren0718 — CHANGES_REQUESTED** [review]

File: core/simple_container.go (hunk 1 above)
Comment (bird512) [reply core/simple_container.go:234]: removed it as suggest

File: core/simple_container.go (hunk 2 above)
Comment (bird512) [reply core/simple_container.go:292]: This is a common problem in our project for now, we should fix all of them in another story

**Pull request merged**